dataset-processor-python/data_loader.py

```diff
@@ -1,55 +1,298 @@
+import json
 import os
+import re
 import sys
 import csv
+import pandas as pd
+import numpy
 
 
-class DataLoader:
+class UserInputHandler:
+    """
+    Class to handle user inputs
+    """
+
     def __init__(self, path, dataset_name):
         self.path = path
         self.dataset_name = dataset_name
-        self.initial_data = None
+        self.inputs = self.get_user_inputs()
+
+    @staticmethod
+    def is_valid_version(version):
+        pattern = r"^[0-9]{4}\.[0-9]+$"
+        return re.match(pattern, version) is not None
+
+    def get_user_inputs(self):
+        version = input("Enter the version: ").strip()
+        if not version or not self.is_valid_version(version):
+            print("Please enter a valid version in the format [yyyy.number].")
+            return self.get_user_inputs()
+        title = input("Enter the title: ").strip()
+        description = input("Enter the description: ").strip()
+        return {
+            "path": self.path,
+            "dataset_name": self.dataset_name,
+            "version": version,
+            "title": title,
+            "description": description,
```

Comment:
maybe in a different PR we could make this a `dataclass` instead of a dictionary.  Also this would benefit from typing.

Also think about what defaults should be.  In case user wants to skip a step, pressing enter should fill in a sensible value if it can.